errors.go

```diff
@@ -17,18 +17,20 @@ import (
 
 // Various errors the driver might return. Can change between driver versions.
 var (
-	ErrInvalidConn       = errors.New("invalid connection")
-	ErrMalformPkt        = errors.New("malformed packet")
-	ErrNoTLS             = errors.New("TLS requested but server does not support TLS")
-	ErrCleartextPassword = errors.New("this user requires clear text authentication. If you still want to use it, please add 'allowCleartextPasswords=1' to your DSN")
-	ErrNativePassword    = errors.New("this user requires mysql native password authentication.")
-	ErrOldPassword       = errors.New("this user requires old password authentication. If you still want to use it, please add 'allowOldPasswords=1' to your DSN. See also https://github.com/go-sql-driver/mysql/wiki/old_passwords")
-	ErrUnknownPlugin     = errors.New("this authentication plugin is not supported")
-	ErrOldProtocol       = errors.New("MySQL server does not support required protocol 41+")
-	ErrPktSync           = errors.New("commands out of sync. You can't run this command now")
-	ErrPktSyncMul        = errors.New("commands out of sync. Did you run multiple statements at once?")
-	ErrPktTooLarge       = errors.New("packet for query is too large. Try adjusting the 'max_allowed_packet' variable on the server")
-	ErrBusyBuffer        = errors.New("busy buffer")
+	ErrInvalidConn          = errors.New("invalid connection")
+	ErrMalformPkt           = errors.New("malformed packet")
+	ErrNoTLS                = errors.New("TLS requested but server does not support TLS")
+	ErrCleartextPassword    = errors.New("this user requires clear text authentication. If you still want to use it, please add 'allowCleartextPasswords=1' to your DSN")
+	ErrNativePassword       = errors.New("this user requires mysql native password authentication")
+	ErrOldPassword          = errors.New("this user requires old password authentication. If you still want to use it, please add 'allowOldPasswords=1' to your DSN. See also https://github.com/go-sql-driver/mysql/wiki/old_passwords")
+	ErrUnknownPlugin        = errors.New("this authentication plugin is not supported")
+	ErrOldProtocol          = errors.New("MySQL server does not support required protocol 41+")
+	ErrPktSync              = errors.New("commands out of sync. You can't run this command now")
+	ErrPktSyncMul           = errors.New("commands out of sync. Did you run multiple statements at once?")
+	ErrPktTooLarge          = errors.New("packet for query is too large. Try adjusting the 'max_allowed_packet' variable on the server")
+	ErrBusyBuffer           = errors.New("busy buffer")
+	ErrNoOptionalResultSet  = errors.New("requested optional resultset metadata but server does not support")
+	ErrOptionalResultSetPkt = errors.New("malformed optional resultset metadata packets")
```

Comment:
Those variable should contain `Metadata` in the name, because that's not the resultset which is optional but just the metadata.